### ISSUE TITLE
Implement Terraform Resource: azuredevops_group

### ISSUE THREAD

**Issue by tmeckel** [microsoft/terraform-provider-azuredevops#103]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Implement a Terraform resource to manage project groups.

### New or Affected Resource(s)

* azuredevops_project_group resource

### Potential Terraform Configuration

```hcl

resource "azuredevops_project" "main_project" {
  project_name       = "Test Project"
  description        = "Test Project Description"
  visibility         = "private"
  enable_tfvc        = False
  work_item_template = "Agile"
}

data "azuredevops_user" "lookup-user" {
   # Exports: id, descriptor
   user_name = "jmarks@vscsi.us" # mapped to: principalName
   origin    = "aad" # Optional (!)
}

data azuredevops_git_repository "lookup-repo" {
    project_id      = azuredevops_project.main_project.id
    repository_name = "Test Lookup Repo"
}

data "azuredevops_group" "lookup-project-group" {
   # Exports: id, descriptor
   # If project_id is omitted the data source will lookup the group inside the organization
   project_id = azuredevops_project.main_project.id # Optional (!)
   group_name = "Test Group"
}

resource "azuredevops_user" "org-user" {
    # Exports: id, descriptor
    origin = "mail" # mail, ad, aad, msa, vsts ...
    id     = "theUesr@mail.com"

    groups = [
        data.azuredevops_group.lookup-project-group.descriptor
    ]
}

resource "azuredevops_group" "organization-group" {
    mail    = "theGroup@mail.com"

    members = [
        data.azuredevops_group.lookup-project-group.descriptor
        azuredevops_user.project-user.descriptor
    ]
}

resource "azuredevops_group" "organization-group-2" {
    origin_id     = "7ffd8466-a183-4948-b1db-a411fac4972a"

    members = [
        data.azuredevops_group.lookup-project-group.descriptor
        azuredevops_user.project-user.descriptor
    ]
}

resource "azuredevops_group" "organization-group-3" {
    origin_id     = "S-1-9-1551374245-1204400969-2402986413-2179408616-4207187754"

    members = [
        data.azuredevops_group.lookup-project-group.descriptor
        azuredevops_user.project-user.descriptor
    ]
}


resource "azuredevops_group" "project-group" {
    # Exports: id, descriptor (Security Descriptor)
    scope                 = azuredevops_project.main_project.descriptor
    displayName           = "Test Group" # displayName
    description           = "Test Description" # Optional (!)

    members = [
        azuredevops_user.project-user.descriptor
        data.azuredevops_user.lookup-project-user.descriptor
        data.azuredevops_group.lookup-project-group.descriptor
        azuredevops_group.organization-group.descriptor
    ]
}

```

### References

* #74 
* #84

* [Groups - Create](https://docs.microsoft.com/en-us/rest/api/azure/devops/graph/groups/create?view=azure-devops-rest-5.1)
* https://github.com/microsoft/azure-devops-go-api/blob/b909a25cb2ee1dc19794e5e8cbdec47dbc139c82/azuredevops/graph/client.go#L33
* [az devops security group](https://docs.microsoft.com/en-us/cli/azure/ext/azure-devops/devops/security/group?view=azure-cli-latest#ext-azure-devops-az-devops-security-group-create)
### Remarks

* The structure `GraphGroupOriginIdCreationContext` seems to lack a property `Origin *string`like the struct `GraphUserOriginIdCreationContext`.

**Comment by nmiodice** [comment]
@tiwood

**Comment by tmeckel** [comment]
As long as it's assigned to @tiwood or me no sweat :-D

**Comment by tmeckel** [comment]
This issue has been implemented with PR #205
